### PR TITLE
contributing.md

### PR DESCRIPTION
**Pull request by Rizzen**
Added Contributing.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+### Checklist before creating a Pull Request
+Submit only relevant commits. We don't mind many commits in a pull request, but they must be relevant as explained below.
+
+- __Use a feature branch__ The pull request should be created from a feature branch, and not from _dev_. See below for why.
+- __Descriptive commit messages__ If a commit's message isn't descriptive, change it using [interactive rebase](https://help.github.com/articles/about-git-rebase). Refer to issues using `#issue`. Example of a bad message ~~"Small cleanup"~~. Example of good message: _"Fixing issue #5 by adding Trie class with supoort of Insert, FindPrefix and Clear and added relevant unit test."_. Don't be afraid to write long messages, if needed. The Erlang repo has some info on [writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages).
+- __No one-commit-to-rule-them-all__ Large commits that changes too many things at the same time are very hard to review. Split large commits into smaller. See this [StackOverflow question](http://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits) for information on how to do this.
```
Comment (admiralWoop) [inline CONTRIBUTING.md]: A typo. "Large commits that change too many things" is the right version.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Contributing
+
+## How to contribute an implementation (code)?
+
+- Have a look at open issues. They contain the list of algorithms/DS we plan
+to be implemented. Pick an unassigned issue.
```
Comment (admiralWoop) [inline CONTRIBUTING.md]: "We are implementing" or "We are planning to implement"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- Feel free to add a new issue for an algorithm that is not in the list.
+- **Make sure you are assigned for the issue.**
+- Implement the algorithm/DS. You are allowed to use any language. See the styleguide below.
+- Always write test on your algorithm or data structure
```
Comment (admiralWoop) [inline CONTRIBUTING.md]: "Always write tests for algorithms/DE you are implementing"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- Make a PR (see checklist below).
+- Be sure to not include any compiled binaries in the patch.
+- While sending a PR make sure you follow one issue per PR rule.
```
Comment (admiralWoop) [inline CONTRIBUTING.md]: What is "PR rule"?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+- __Use a feature branch__ The pull request should be created from a feature branch, and not from _dev_. See below for why.
+- __Descriptive commit messages__ If a commit's message isn't descriptive, change it using [interactive rebase](https://help.github.com/articles/about-git-rebase). Refer to issues using `#issue`. Example of a bad message ~~"Small cleanup"~~. Example of good message: _"Fixing issue #5 by adding Trie class with supoort of Insert, FindPrefix and Clear and added relevant unit test."_. Don't be afraid to write long messages, if needed. The Erlang repo has some info on [writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages).
+- __No one-commit-to-rule-them-all__ Large commits that changes too many things at the same time are very hard to review. Split large commits into smaller. See this [StackOverflow question](http://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits) for information on how to do this.
+- __Tests__ Add relevant tests and make sure all existing ones still passes. Tests can be run using the command
```
Comment (admiralWoop) [inline CONTRIBUTING.md]: "still pass"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- __No Warnings__ Make sure your code do not produce any build warnings.
+
+## Common Coding Style
+
+- Code submitted should be modular.
```
Comment (admiralWoop) [inline CONTRIBUTING.md]: "submitted code"

File: CONTRIBUTING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rizzen) [reply CONTRIBUTING.md]: "One issue per one PR" rule

[the remark this answers is not in the document]
Comment (admiralWoop) [reply CONTRIBUTING.md]: Thanks

**Pull request merged**